osh/builtin_meta.py

```diff
@@ -75,66 +77,81 @@ def Run(self, cmd_val):
 
 class Source(vm._Builtin):
     def __init__(self, parse_ctx, search_path, cmd_ev, fd_state, tracer,
-                 errfmt):
-        # type: (ParseContext, state.SearchPath, CommandEvaluator, process.FdState, dev.Tracer, ui.ErrorFormatter) -> None
+                 errfmt, loader):
+        # type: (ParseContext, state.SearchPath, CommandEvaluator, process.FdState, dev.Tracer, ui.ErrorFormatter, pyutil._ResourceLoader) -> None
         self.parse_ctx = parse_ctx
         self.arena = parse_ctx.arena
         self.search_path = search_path
         self.cmd_ev = cmd_ev
         self.fd_state = fd_state
         self.tracer = tracer
         self.errfmt = errfmt
+        self.loader = loader
 
         self.mem = cmd_ev.mem
 
     def Run(self, cmd_val):
         # type: (cmd_value.Argv) -> int
-        call_loc = cmd_val.arg_locs[0]
-        _, arg_r = flag_spec.ParseCmdVal('source', cmd_val)
+        attrs, arg_r = flag_spec.ParseCmdVal('source', cmd_val)
+        arg = arg_types.source(attrs.attrs)
 
         path = arg_r.Peek()
         if path is None:
             e_usage('missing required argument', loc.Missing)
         arg_r.Next()
 
-        resolved = self.search_path.Lookup(path, exec_required=False)
-        if resolved is None:
-            resolved = path
+        if arg.builtin:
+            contents = self.loader.Get(path)
```

Comment:
I think we should catch the IOError and print a message, and return error 2

Right now it looks like

```
$ bin/ysh -c 'source --builtin stdlib/math.ys'
oils I/O error (main): No such file or directory
```

( I actually think it would be nicer if loader.Get() returns None not IOError ... i.e. `Optional[str]`  ... that was left over from the file interface )